Userland/Libraries/LibELF/DynamicLoader.cpp

```diff
@@ -448,6 +442,15 @@ void DynamicLoader::load_program_headers()
     quick_sort(map_regions, compare_load_address);
     quick_sort(copy_regions, compare_load_address);
 
+    // Finally, we unmap the reservation.
+    if (munmap(reservation, total_mapping_size) < 0) {
+        perror("munmap reservation");
+        VERIFY_NOT_REACHED();
+    }
+
+    // WARNING: Allocating after this point has the possibility of malloc stealing our reserved
+    // virtual memory addresses. Be careful not to malloc here!
```

Comment:
We still have `m_text_segments.append` and `StringBuilder` in there, both of which should be trivial to preallocate. In the case of `m_text_segments` we probably just want to `ensure_capacity(map_regions.size())`, and the `StringBuilder` ends up being two different possible strings for the entire binary.

Reply:
good point! I caught a final allocation for the data segment in an inline DeprecatedString::formatted() as well.